File: Janhavi_Tripathi/day11.cpp
/*Using 5 variables.
Here we are adding previous value with the current next element with serial of a-e-i-o-u.
Return by adding every element.
Time complexity: O(n).*/

#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int countVowelStrings(int n) {
        int a=1,e=1,i=1,o=1,u=1;
        while(--n)
        {
            o+=u;
            i+=o;
            e+=i;
            a+=e;
        }
        return a+e+i+o+u;
    }
};

File: Shashank Shukla 18-CSE/Day-11.java
/*
Time: O(N)

*/
class Solution {
    public int countVowelStrings(int n) {
        int[] dp = new int[]{0,1,1,1,1,1};
        for(int i=1; i<=n; i++){
            for(int j=1; j<=5; j++){
                dp[j] += dp[j-1];
            }
        }
        return dp[5];
    }
}
